processor/metric/payload.go

```diff
@@ -0,0 +1,300 @@
+package metric
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/apm-server/config"
+	m "github.com/elastic/apm-server/model"
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+var (
+	transformations = monitoring.NewInt(metricMetrics, "transformations")
+	processorEntry  = common.MapStr{"name": processorName, "event": docType}
+
+	errMissingKey = errors.New("key not found")
+	errNan        = errors.New("not a json number")
+)
+
+type Sample interface {
+	Name() string
+	Value() common.MapStr
+}
+
+func floatFrom(key string, m map[string]interface{}) (float64, error) {
```

Comment:
Why not using `utility/data_fetcher` here? https://github.com/elastic/apm-server/blob/master/utility/data_fetcher.go#L19

Reply:
Probably because I forgot it existed, I'll rework this.